File: PHP/leetcode/2-keys-keyboard/solution.php
class Solution {

    /**
     * @param Integer $n
     * @return Integer
     */
    function minSteps($n) {
        $res = 0;
        for ($d = 2; $d <= $n; $d++) {
            while ($n % $d == 0) {
                $res += $d;
                $n = intdiv($n, $d);
            }
        }
        return $res;
    }
}

File: Kotlin/leetcode/2-keys-keyboard/solution.kt
class Solution {
    fun minSteps(n: Int): Int {
        var res = 0
        var m = n
        var d = 2
        while (d <= m) {
            while (m % d == 0) { res += d; m /= d }
            d++
        }
        return res
    }
}
